### ISSUE TITLE
Inferencing code on the reward model

### ISSUE THREAD

**Issue by Lsayan** [RamyaLab/pluralistic-alignment#5]
Hi @ChenDaiwei-99, thanks for sharing the work.
I wish to know that after getting the reward model, can you please share the inferencing code on the reward model. 

Basically, I wanted to know the format in which the input need to be passed to the reward model and what should the input contain? (Prompt, Answer1, Answer2) ?

And can you please tell what the reward model is supposed to output? 
I was running cell 21 of /integration/pal_integration.ipynb and it returned a set of tensors (as in the notebook shared) . Can you please clarify which is the reward score here?
In cell 20 of /integration/pal_integration.ipynb, you are only passing the input_ids and attention_mask of Prompt and Chosen as input. What does the reward score signify here?

Thanks in advance for the response.

Regards
Sayan

**Comment by Lsayan** [comment]
@ChenDaiwei-99 Can you please share the inferencing code  for the reward model that is created using pal_b?

**Comment by ChenDaiwei-99** [comment]
Q1. Basically, I wanted to know the format in which the input need to be passed to the reward model and what should the input contain? (Prompt, Answer1, Answer2) ?

_A1. Please check the section `#Integration` in the README.md file._

---

Q2. And can you please tell what the reward model is supposed to output?

_A2. The output of the reward would be a scalar. Basically, the reward model is $r(s,a): \mathcal{S}\times\mathcal{A}\to\mathbb{R}$._

---

Q3. I was running cell 21 of /integration/pal_integration.ipynb, and it returned a set of tensors (as in the notebook shared) . Can you please clarify which is the reward score here?

_A3. Sorry about the ambiguous and not well-documented code in the `/integration` subfolder. My Bad :(. The returned tensor represents the reward for each token ( equivalent to the reward for the partial sequence). ( Yes, we implicitly did the token-wise reward modeling! ). But if you don't care about the token-wise reward, you can only use the last token's reward, which is exactly the sequence-level reward, as the typical reward model does._

---

Q4. In cell 20 of /integration/pal_integration.ipynb, you are only passing the input_ids and attention_mask of Prompt and Chosen as input. What does the reward score signify here?

_A4. Once you've trained the PAL reward model and converted it into a standard reward model formula, you can pass any (prompt, response) pair into the reward model and get the corresponding reward, as the typical reward model does._

---

Q5. Can you please share the inferencing code for the reward model that is created using pal_b?

_A5. I highly recommend you to check our open-sourced [pal-b-large-opt-350m](https://huggingface.co/daiweichen/pal-b-large-opt-350m/tree/main) model on HuggingFace for details about the inferencing code._

---
